validation/jinja2/validation/view_issue_detail.html

```diff
@@ -19,6 +19,24 @@
 {% import 'validation/_macros.html' as macros %}
 
 {% block content %}
+    {% if issue.phrase %}
+        <script>
+            document.addEventListener('DOMContentLoaded', () => {
+                let transcription = document.getElementById("id_transcription");
+                let translation = document.getElementById("id_translation");
+                transcription.value = "{{ issue.suggested_cree }}"
+                translation.value = "{{ issue.suggested_english }}"
+            })
+        </script>
+    {% elif issue.recording %}
+        <script>
+            document.addEventListener('DOMContentLoaded', () => {
+                let phrase = document.getElementById("id_phrase");
+                phrase.value = "{{ issue.suggested_cree }}"
+            })
+        </script>
+    {% endif %}
+
```

Comment:
This has to be done inline, otherwise it doesn't have access `issue` and fills in the literal string "{{ issue.etc }}". I don't love this solution, but it gets the job done. I'm more than willing to brainstorm a better way of accomplishing this.

Reply:
[Form initial data](https://docs.djangoproject.com/en/3.2/ref/forms/api/#initial-form-values) will only be used if the form is not ‘bound’, i.e., if it appears to be a brand-new empty form created only by code, and not one already incorporating some user input.

[The current code in the `view_issue_detail` view](https://github.com/UAlbertaALTLab/recording-validation-interface/blob/b7948e304abbf4057fb221c5dca8cec62539f54c/validation/views.py#L548) does `form = EditIssueWithRecording(request.POST)` regardless of whether there is any POST data or not, meaning that django thinks this is never a brand-new form but always a user submission, which is why the `initial` value gets ignored.

I think this would be a more fruitful direction to go in:

```
def view_issue_detail(request, issue_id):
    issue = Issue.objects.get(id=issue_id)

    if issue.recording:
        if request.method == "POST":
            form = EditIssueWithRecording(request.POST)
            if form.is_valid():
                return handle_save_issue_with_recording(form, issue, request)
        else:
            form = EditIssueWithRecording(initial={
                'phrase': issue.comment
            })

    if issue.phrase:
        if request.method == 'POST':
            form = EditIssueWithPhrase(request.POST)
            if form.is_valid():
                return handle_save_issue_with_phrase(form, issue, request)
        else:
            form = EditIssueWithPhrase(initial={
                'transcription': issue.suggested_cree,
                'translation': issue.suggested_english
            })

    context = dict(
        issue=issue,
        form=form,
        auth=request.user.is_authenticated,
        is_linguist=user_is_linguist(request.user),
    )
    return render(request, "validation/view_issue_detail.html", context)
```

Reply:
Thank you for figuring this out! This is so much better than the JS.